libimage/search.go

```diff
@@ -235,7 +238,11 @@ func (r *Runtime) searchImageInRegistry(ctx context.Context, term, registry stri
 			authConf.Password = split[1]
 		}
 	}
-	sys.DockerAuthConfig = authConf
+	// We should set the authConf unless a token was set.  That's especially
+	// useful for Podman's remote API.
+	if options.IdentityToken != "" {
```

Comment:
- Doesn’t this completely break uses of non-`IdentityToken` credentials?
- I don’t understand what this condition is trying to express at all, and how it relates to the comment. If a token was set, we AFAICS need to set `authConf` as well, as this code does — but the comment says it doesn’t.
- I also don’t understand how, if at all, it relates to the remote API; AFAICS remote/local should make no difference to the general structure of this code.
- I think it would be more reliable for callers, and also allow a clearer expression of this code, if the code enforced that only _one_ source of credentials were used.

```go
numCredsSources := 0
if options.Username != "" { numCredsSources++; … }
if options.Credentials != "" { numCredsSources++; … }
if options.IdentityToken != "" { numCredsSources++; … }
if numCredsSources > 1 { /* error */ }
// possibly decide based on numCredsSources != 0
```

Reply:
> Doesn’t this completely break uses of non-IdentityToken credentials?

Thank you for your reply.
I verified that search with `AuthFilePath` option is failed if `non-IdentityToken` credentials.
I'll fix it to use getDockerAuthConfig() such as Pull.